system/dns.go

```diff
@@ -23,40 +28,64 @@ type DefDNS struct {
 	Timeout     int
 	loaded      bool
 	err         error
+	server      string
+	qtype       string
 }
 
 func NewDefDNS(host string, system *System, config util.Config) DNS {
+	var h string
+	var t string
+	if len(strings.Split(host, ":")) > 1 {
+		h = strings.Split(host, ":")[1]
+		t = strings.Split(host, ":")[0]
+	} else {
+		h = host
+	}
 	return &DefDNS{
-		host:    host,
+		host:    h,
 		Timeout: config.Timeout,
+		server:  config.Server,
+		qtype:   t,
 	}
 }
 
 func (d *DefDNS) Host() string {
 	return d.host
 }
 
+func (d *DefDNS) Server() string {
+	return d.server
+}
+
+func (d *DefDNS) Qtype() string {
+	return d.qtype
+}
+
 func (d *DefDNS) setup() error {
 	if d.loaded {
 		return d.err
 	}
 	d.loaded = true
 
-	addrs, err := lookupHost(d.host, d.Timeout)
-	if err != nil || len(addrs) == 0 {
-		d.resolveable = false
-		d.addrs = []string{}
-		// DNSError is resolvable == false, ignore error
-		if _, ok := err.(*net.DNSError); ok {
-			return nil
+	for i := 0; i < 3; i++ {
+		addrs, err := DNSlookup(d.host, d.server, d.qtype, d.Timeout)
+		if err != nil || len(addrs) == 0 {
```

Comment:
That awkward moment when you don't recognize your own code..

Anyways, this looks good to me, CI passing, did some local testing and it's looking good. If this is good to go on your end, I'll merge it in.